src/playlist.js

```diff
@@ -541,6 +542,74 @@ export const isLowestEnabledRendition = (master, media) => {
   }).length === 0);
 };
 
+const playlistMatch = (a, b) => {
+  if (a.id && b.id && a.id === b.id) {
+    return true;
+  }
+
+  if (a.uri && b.uri && a.uri === b.uri) {
+    return true;
+  }
+
+  if (a.resolvedUri && b.resolvedUri && a.resolvedUri === b.resolvedUri) {
```

Comment:
Can we have a `resolvedUri` without a `uri`?

Reply:
we can't I am going to move the resolvedUri check above the uri check and add some comments.